src/Misc/XMLwrapper.cpp

```diff
@@ -204,7 +239,14 @@ char *XMLwrapper::getXMLdata() const
 {
     xml_k = 0;
 
+#if MXML_MAJOR_VERSION <= 3
     char *xmldata = mxmlSaveAllocString(tree, XMLwrapper_whitespace_callback);
+#else
+    mxml_options_t *options = mxmlOptionsNew();
```

Comment:
Sanity check, but I see a few instances with mxmlOptionsNew() and based on similar C APIs I'd worry that we'd be responsible for deallocating the returned mxml_options_t pointer. Can you verify if we need to deallocate that value or if my concern is unfounded?